File: JMMinistry.Common/Dtos/User/UserInfoDto.cs
using JMMinistry.Common.Dtos.User.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace JMMinistry.Common.Dtos.User
{
    public class UserInfoDto: PartialUserInfoDto
    {
        public string City { get; set; } = null!;

        public string? Locality { get; set; }

        public string Neighborhood { get; set; } = null!;

        public string Address { get; set; } = null!;
        public string Email { get; set; } = string.Empty;
        public string Profession { get; set; } = string.Empty;
        public string Occupation { get; set; } = string.Empty;

        public DateOnly Birthday { get; set; }

        public MaritalStatus? MaritalStatus { get; set; }
        public EducationalLevel? EducationalLevel { get; set; }

        public AccessType AccessType { get; set; }

        public List<PartialUserInfoDto> Leaders { get; set; } = [];
    }
}

File: the_vine_hub_app/lib/shared/domain/models/user_info_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:the_vine_hub_app/shared/domain/models/partial_user_info_dto.dart';

part 'user_info_dto.freezed.dart';
part 'user_info_dto.g.dart';

@freezed
abstract class UserInfoDto with _$UserInfoDto {
  const UserInfoDto._();

  const factory UserInfoDto({
    String? document,
    String? name,
    String? lastName,
    String? phone,
    int? gender,
    int? maritalStatus,
    String? photo,
    int? cellId,
    String? email,
    String? address,
    String? city,
    String? locality,
    String? neighborhood,
    String? profession,
    String? occupation,
    DateTime? birthday,
    int? educationalLevel,
    int? accessType,
    List<PartialUserInfoDto>? leaders,
  }) = _UserInfoDto;

  factory UserInfoDto.fromJson(Map<String, dynamic> json) => _$UserInfoDtoFromJson(json);
}
